account_payment_order/models/account_payment_order.py

```diff
@@ -196,6 +196,12 @@ def check_date_scheduled(self):
                         )
                     )
 
+    @api.constrains("payment_line_ids")
+    def _check_payment_lines(self):
+        for order in self:
+            if len(order.payment_line_ids.move_line_id) != len(order.payment_line_ids):
```

Comment:
This constraint will break payment orders with payment lines without journal items. The journal item is not mandatory on the payment line. If a payment line is added manually without journal item, the constraint will be triggered.